src/interchange/src/protobuf.rs

```diff
@@ -125,10 +125,25 @@ fn validate_proto_field_resolved<'a>(
     Ok(())
 }
 
-pub fn decode_descriptors(descriptors: &[u8]) -> Result<Descriptors> {
-    let proto = protobuf::Message::parse_from_bytes(descriptors)
-        .context("parsing encoded protobuf descriptors failed")?;
-    Ok(Descriptors::from_proto(&proto))
+pub fn decode_descriptors(descriptors: &[u8]) -> Result<DecodedDescriptors> {
```

Comment:
Could you document (in that same future pr) that this always uses the first descriptor in the file to match how confluent tools do it?

Reply:
Yes! Thank you, forgot that part!